src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcGroupingAnalyzers.scala

```diff
@@ -1,124 +0,0 @@
-/**
-  * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
-  *
-  * Licensed under the Apache License, Version 2.0 (the "License"). You may not
-  * use this file except in compliance with the License. A copy of the License
-  * is located at
-  *
-  *     http://aws.amazon.com/apache2.0/
-  *
-  * or in the "license" file accompanying this file. This file is distributed on
-  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
-  * express or implied. See the License for the specific language governing
-  * permissions and limitations under the License.
-  *
-  */
-
-package com.amazon.deequ.analyzers.jdbc
-
-import java.sql.ResultSet
-
-import JdbcAnalyzers._
-import Preconditions._
-import com.amazon.deequ.metrics.DoubleMetric
-
-/** Base class for all analyzers that operate the frequencies of groups in the data */
-abstract class JdbcFrequencyBasedAnalyzer(columnsToGroupOn: Seq[String])
-  extends JdbcAnalyzer[JdbcFrequenciesAndNumRows, DoubleMetric] {
-
-  def groupingColumns(): Seq[String] = { columnsToGroupOn }
-
-  override def computeStateFrom(table: Table): Option[JdbcFrequenciesAndNumRows] = {
-    Some(JdbcFrequencyBasedAnalyzer.computeFrequencies(table, groupingColumns()))
-  }
-
-  /** We need at least one grouping column, and all specified columns must exist */
-  override def preconditions: Seq[Table => Unit] = {
-    Seq(hasTable()) ++ Seq(atLeastOne(columnsToGroupOn)) ++
-      columnsToGroupOn.map { hasColumn } ++ super.preconditions
-  }
-}
-
-object JdbcFrequencyBasedAnalyzer {
-
-  /** Compute the frequencies of groups in the data, essentially via a query like
-    *
-    * SELECT colA, colB, ..., COUNT(*)
-    * FROM DATA
-    * WHERE colA IS NOT NULL AND colB IS NOT NULL AND ...
-    * GROUP BY colA, colB, ...
-    */
-  def computeFrequencies(
-    table: Table,
-    groupingColumns: Seq[String],
-    numRows: Option[Long] = None)
-  : JdbcFrequenciesAndNumRows = {
-
-    val connection = table.jdbcConnection
-    // TODO: allow multi column
-    val column = groupingColumns.head
-
-    val query =
-      s"""
-         | SELECT $column as name, COUNT(*) AS absolute
```

Comment:
Yes currently we support only one column because our `JdbcFrequenciesAndNumRows` is not as flexible as the `dataframe` of spark. Later on it will be one query per metric but for all columns.

Reply:
Is that what you mean or do you mean some other kind of reduction?